flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -3034,6 +3043,54 @@ IntrinsicLibrary::genSize(mlir::Type resultType,
   return builder.createConvert(loc, resultType, result);
 }
 
+// LBOUND
+fir::ExtendedValue
+IntrinsicLibrary::genLbound(mlir::Type resultType,
+                            llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 3);
+  if (const auto *boxValue = args[0].getBoxOf<fir::BoxValue>())
+    if (boxValue->hasAssumedRank())
+      TODO(loc, "LBOUND intrinsic with assumed rank argument");
+
+  // Calls to LBOUND that don't have the DIM argument are handled, or for which
+  // the DIM is a compile time constant are folded to descriptor inquiries by
+  // semantics.
```

Comment:
I don't understand this comment.  Can you please try rewording it?

Reply:
Rephrased to:
```
   // Calls to LBOUND that don't have the DIM argument, or for which
   // the DIM is a compile time constant, are folded to descriptor inquiries by
   // semantics
```

Reply:
Looks perfect!